src/xkb_bell.c

```diff
@@ -76,8 +79,10 @@ void initialize_xkb(void) {
 		return;
 	}
 
-	if (! XkbOpenDisplay(DisplayString(dpy), &xkb_base_event_type, &ir,
-	    NULL, NULL, &reason) ) {
+	xkb_disp = XkbOpenDisplay(DisplayString(dpy), &xkb_base_event_type, &ir,
+		NULL, NULL, &reason);
+
+	if (!xkb_disp) {
```

Comment:
I don't think I'm sure what you mean by that... If ```XkbOpenDisplay``` fails, ```xkb_disp``` is ```NULL``` anyway. The connection needs to stay open until x11vnc is terminated, does it? Or could it be closed right away in any case?